overlord/snapstate/snapmgr.go

```diff
@@ -257,7 +259,12 @@ func readInfo(name string, si *snap.SideInfo, flags int) (*snap.Info, error) {
 		if si != nil {
 			info.SideInfo = *si
 		}
-		return info, nil
+		err = nil
+	}
+	if err == nil && flags&withAuxStoreInfo != 0 {
```

Comment:
Is there a downside (except that it takes one extra disk-read) for loading the aux store info all the time?

Reply:
very small consideration: it'd be a little bit slower when not needed.
Very big consideration: it'd break so many tests.

Reply:
I'm still ok with the flag to be clear